libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts

```diff
@@ -388,6 +402,22 @@ export class WorkspaceInstallManager extends BaseInstallManager {
     return { shrinkwrapIsUpToDate, shrinkwrapWarnings };
   }
 
+  private _getPackageExtensionChecksum(
+    packageExtensions: Record<string, unknown> | undefined
+  ): string | undefined {
+    const packageExtensionsChecksum: string | undefined =
+      Object.keys(packageExtensions ?? {}).length === 0
+        ? undefined
+        : createObjectChecksum(packageExtensions!);
+
+    function createObjectChecksum(obj: Record<string, unknown>): string {
+      const s: string = JSON.stringify(Sort.sortKeys(obj, { deep: true }));
```

Comment:
This might be better to do the sorting as the replacer function to `JSON.stringify`. That would result in less memory usage because it would be able to process only one object or array at a time, rather than having to create a full recursive copy.

Reply:
That would diverge from the implementation in pnpm and I'm not confident in implementing it that way...